app/gui2/src/stores/graph/graphDatabase.ts

```diff
@@ -140,7 +143,22 @@ export class GraphDb {
     function* allTargets(db: GraphDb): Generator<[ExprId, ExprId]> {
       for (const usage of info.usages) {
         const targetNode = db.getExpressionNodeId(usage)
-        // Display only connections to existing targets and different than source node
+        // Display only connections to existing targets and different than source node.
+        if (targetNode == null || targetNode === srcNode) continue
+        yield [alias, usage]
+      }
+    }
+    return Array.from(allTargets(this))
+  })
+
+  /** Same as {@link GraphDb.connections}, but also includes connections without source node,
+   * e.g. input arguments of the collapsed function.
+   */
+  allConnections = new ReactiveIndex(this.bindings.bindings, (alias, info) => {
+    const srcNode = this.getPatternExpressionNodeId(alias)
+    function* allTargets(db: GraphDb): Generator<[ExprId, ExprId]> {
+      for (const usage of info.usages) {
+        const targetNode = db.getExpressionNodeId(usage)
```

Comment:
A repetitive code. Let's make an inner function for getting connections from binding info.

Reply:
Done